python/cudf/cudf/core/_base_index.py

```diff
@@ -629,12 +629,18 @@ def union(self, other, sort=None):
             common_dtype = cudf.utils.dtypes.find_common_type(
                 [self.dtype, other.dtype]
             )
-            return self._get_reconciled_name_object(other).astype(common_dtype)
+            res = self._get_reconciled_name_object(other).astype(common_dtype)
+            if sort is True:
```

Comment:
I think using `if sort` here is fine:

1. `sort` can only be `None`, `False` and `True`. `if sort` only matches `True`.
2. In the 2 shortcut paths modified here, only `sort=True` means sorting.

Reply:
Yup, agree. Made the change 👍